### PR TITLE
Initial work toward a Docker image. Issue #25 

### PR DESCRIPTION
**Pull request by jefferya**
Initial PR for a development and production Docker image, Docker Compose images for trying together [Swift](https://github.com/ualbertalib/di_docker_swift) and Pushmi-Pullyu and Documentation.

Wanting to create PR before the end of sprint and I'm away.

All changes within the `docker` directory off repo root

Short comings:
* see **warnings** in docker/README.md for [details](https://github.com/ualbertalib/pushmi_pullyu/tree/issue_25_docker_image/docker#future-considerations): summary provided here
  * /etc/redis/redis.conf bind only open to localhost on the HydraNorth Docker image creates the need for a work-around [reference](https://github.com/ualbertalib/di_docker_hydranorth/issues/12)
  * assumption in rspec code that Redis runs on localhost (ignores config file) thus env variable REDIS_URL required 
* Unsure how to update DockerHub
* Unsure how to build production and test components not yet merged into master

### CODE REVIEW

File: docker/.dockerignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.git
```
Comment (murny) [inline docker/.dockerignore]: This is great you added a `.dockerignore` as none of our ualbertalib images currently has this. So great job :+1: .

However we should add a few more things to this:
```
log/*
coverage
.travis.yml
pkg
.bundle
*.md
docs
```

Maybe bin folder too?

**Comment by weiweishi** [comment]
Is this PR mergable, or is it blocked by the Redis issue? @murny

**Comment by murny** [comment]
Blocked by all the issues labelled with `docker`.  But this is a nice to have and not really urgent. Low pirority.

After Hydranorth docker is cleaned up and pulled apart, this will make this all super simple. I am fine with closing this PR for time being? Especially once hydranorth docker work is complete, half this PR won't be needed.

**Comment by cwant** [comment]
Closing based on advice from Shane and Jeff.

**Pull request closed without merging**